esp-hal/src/i2c.rs

```diff
@@ -370,7 +370,7 @@ where
         sda: impl Peripheral<P = SDA> + 'd,
         scl: impl Peripheral<P = SCL> + 'd,
         frequency: HertzU32,
-        clocks: &Clocks<'d>,
```

Comment:
Is this correct? Won't the lifetime be inferred to something shorter that it should be?

Reply:
I think it is, but regardless this change shouldn't be part of this PR.